### ISSUE TITLE
Unmaintained? 

### ISSUE THREAD

**Issue by gagarine** [au2001/icloud-passwords-firefox#52]
This plugin appears to be unmaintained, with the last commit made five months ago.

Given the security sensitivity of this plugin, we would greatly appreciate any clarification on its current status.

Thank you!

**Comment by au2001** [comment]
Hi, Apple is going to release an official version soon
I will see if it's worth it to continue working on new features then
In the mean time, security fixes are still definitely taken seriously

**Comment by romanmgelman** [comment]
Source? I really appreciate all the work you and contributors have done for us Mac Firefox users!

**Comment by gagarine** [comment]
Thank you for the info! Im glad to hear securities issue are still treated. Le 28 oct. 2024 à 19:32, Roman Gelman ***@***.***> a écrit :﻿
Source? I really appreciate all the work you and contributors have done for us Mac Firefox users!

**Comment by au2001** [comment]
> Source?

![Screenshot 2024-10-28 at 20 49 56](https://github.com/user-attachments/assets/a7474212-dfb8-4f6e-9e2c-02481fbb54b6)

**Comment by romanmgelman** [comment]
merci!

**Comment by regulskimichal** [comment]
Based on the data from webarchive, it looks to me like it has been handed over to Apple.
On the other hand, it is relatively easy to adapt their CRX from Chrome to XPI. As a downside, if anyone will try it on their own:
- change `manifest.json` by removing incompatible properties and switching from
```json
"background": {
  "service_worker": "background.js"
}
```
to
```json
"background": {
  "scripts": ["background.js"]
}
```
- add extension id to `manifest.json`:
```
"browser_specific_settings": {
  "gecko": {
    "id": "password-manager-firefox-extension@apple.com"
  }
}
```
- they need to setup NativeMessagingHosts according to the platform, or use scripts from `scripts/installation` folder
- load extension to Firefox Developer Edition/ESR with disabled feature-flag `xpinstall.signatures.required`

**Comment by au2001** [comment]
> @the-gray-ghost Any idea why your add-on is now showing up as the official Apple one in the Firefox Add-On store?

Quoting from [Reddit](https://www.reddit.com/r/firefox/comments/1gorijc/comment/lwkomyj/):

Yes Apple did reach out to me through Mozilla
They plan on releasing an official version soon™
For technical reasons, my extension is the only one which can read iCloud Passwords
So I accepted to transfer it to them

The source code is still available on GitHub, and if Apple's version isn't good enough I will re-publish it under a different listing and continue to maintain it
Otherwise I'll update the GitHub readme and archive the repository

**Comment by au2001** [comment]
> @regulskimichal On the other hand, it is relatively easy to adapt their CRX from Chrome to XPI.

Manually porting the Chrome extension is not necessary
The (now official) extension on AMO still works fine

Although your steps should also work for people who do not wish to trust the code from this repository

**Comment by regulskimichal** [comment]
I missed one feature - 2FA codes, although I saw your branch and I truly appreciate your effort to make iCloud Passwords for everyone

**Comment by biotomed** [comment]
I tried to automatically (and manually) update the registry and json files as seen in your install scripts to get the plug-in working for Windows with the new iCloud app, but unfortunately that didn’t work. Is there no way right now to get it working for windows? How long do you think I have to wait until Apple or you release a working option? :)

**Comment by leftbones** [comment]
I'm hoping the official one doesn't require entering a code to make it work, I was surprised to see that the same thing is required for the Chrome extension, which I thought was official.

**Comment by au2001** [comment]
> Is there no way right now to get it working for windows?

There is no documented way right now unfortunately

> How long do you think I have to wait until Apple or you release a working option? :)

I don't have much info on their timeline either

> I'm hoping the official one doesn't require entering a code to make it work, I was surprised to see that the same thing is required for the Chrome extension, which I thought was official.

The Chrome and Edge extensions are official
They require entering a code to avoid other apps reading your accounts
I do not see Apple changing this behavior any time soon, but that's just my opinion
The most probable option IMO would be a first-party iCloud Keychain integration in Firefox itself one day

**Comment by 101Dude** [comment]
I suggest no one install this until it's 100% clear who owns the page on [Mozilla](https://addons.mozilla.org/en-US/firefox/user/10200780/).

The Mozilla Plugin page for this is very phishy

Reads:

> iCloud Passwords by [Apple Inc.](https://addons.mozilla.org/en-US/firefox/user/10200780/)

and also says:

> This extension is not endorsed in any way by Apple Inc.
> iCloud, iCloud Keychain, and macOS are trademarks of Apple Inc., registered in the U.S. and other countries and regions.

Even more odd is why Apple would produce iCloud Keychain when they now are integrating their Passwords.app functionality into browsers.

**Comment by m1ten** [comment]
> and also says:
> 
> > This extension is not endorsed in any way by Apple Inc.
> > iCloud, iCloud Keychain, and macOS are trademarks of Apple Inc., registered in the U.S. and other countries and regions.

I’m quite certain that’s inherited from the initial listing. They haven’t made any updates since then, so in a sense, it’s “not endorsed.”

**Comment by hakusaro** [comment]
> Even more odd is why Apple would produce iCloud Keychain when they now are integrating their Passwords.app functionality into browsers.

@101Dude actually, this is the way that Apple usually integrates with third-party browsers. They just publish an extension that can talk to the official Passwords app.

@au2001 thank you for all your hard work, and I really hope Apple at least paid you a little for this.

**Comment by au2001** [comment]
> I suggest no one install this until it's 100% clear who owns the page on [Mozilla](https://addons.mozilla.org/en-US/firefox/user/10200780/).
> 
> The Mozilla Plugin page for this is very phishy

Apple owns the listing, I own the code

If you do not trust the code, then you should wait for Apple to release their own

> I’m quite certain that’s inherited from the initial listing. They haven’t made any updates since then, so in a sense, it’s “not endorsed.”

Yes, precisely

> @101Dude actually, this is the way that Apple usually integrates with third-party browsers. They just publish an extension that can talk to the official Passwords app.

Right, Apple can only integrate their new Passwords.app through an (my) extension, like they do on Chrome or for iCloud Bookmarks

Having first-party integration in Firefox requires Mozilla to do it, or at least Apple to work with them

> @au2001 thank you for all your hard work, and I really hope Apple at least paid you a little for this.

Unfortunately not, I suggested a $1 one-time GitHub sponsorship as acknowledgement before the transfer but didn't get an answer

**Comment by hakusaro** [comment]
> Unfortunately not, I suggested a $1 one-time GitHub sponsorship as acknowledgement before the transfer but didn't get an answer

Typical. I'm sorry but I don't have a lot of free cash flow (I just moved), but please enjoy a small thankyou, because I really appreciate this extension!

**Comment by ctrlMarcio** [comment]
Well, now it simply doesn't work (at least for me)

> An unknown error occurred. Make sure you have the latest version of the extension installed. If the error persists, please try restarting Firefox.

IDK if it's because of the new FF version, but I guess it won't be updated.

**Comment by ctrlMarcio** [comment]
> Are you using Windows perhaps? Sounds like https://github.com/au2001/icloud-passwords-firefox/issues/40

Yep, sorry. Answer in #38 worked out for me.

**Comment by MrMcFly2k** [comment]
Finally, Apple released its **official** iCloud Passwords Firefox extension:

https://addons.mozilla.org/en-US/firefox/addon/icloud-passwords/

**Comment by ayndqy** [comment]
Thanks for your work! It was a pleasure to use your extension until the official one came out.

**Comment by MrMcFly2k** [comment]
@au2001 could you please confirm if it has been handed over to Apple?

https://addons.mozilla.org/en-US/firefox/addon/icloud-passwords/

**Comment by au2001** [comment]
> @au2001 could you please confirm if it has been handed over to Apple?

Yes, Apple is now the sole owner/maintainer of the iCloud Passwords extension on Firefox
Their version is not linked in any way to this repo anymore

**Comment by minhquangngo** [comment]
> Yes, Apple is now the sole owner/maintainer of the iCloud Passwords extension on Firefox Their version is not linked in any way to this repo anymore

Are there any official confirmations from coming from Apple? I only see Chromium browsers listed as the official extension.

**Comment by hakusaro** [comment]
@minhquangngo you can refer to the official Apple Inc., page on Firefox’s Addons store: https://addons.mozilla.org/en-US/firefox/user/10200780/

**Comment by au2001** [comment]
> Are there any official confirmations from coming from Apple?

Not that I know of. Closest you'll get is [an Apple employee in charge of Passwords posting from their personal account](https://bsky.app/profile/rmondello.com/post/3ldo2nw4oks2z).
It seems like Apple is staying silent on Firefox for reasons unknown to me.

**Comment by au2001** [comment]
On another note, Apple's official version slowing down my browser and generally being unusable, I have republished mine here:
https://addons.mozilla.org/en-US/firefox/addon/keychain/

I will thus close this issue and update the repository to reflect this.

**Comment by minhquangngo** [comment]
> > Are there any official confirmations from coming from Apple?
> 
> Not that I know of. Closest you'll get is [an Apple employee in charge of Passwords posting from their personal account](https://bsky.app/profile/rmondello.com/post/3ldo2nw4oks2z). It seems like Apple is staying silent on Firefox for reasons unknown to me.

The extension on FF extension store is legit. It is recommended by Apple when you go to the Password app and click on "Get Browser Extension". Thanks @au2001 for making this!
